File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/guardosphere/GuardoSphereButton.java
package de.limbusdev.guardianmonsters.fwmengine.guardosphere;

import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Skin;
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;

import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian;
import de.limbusdev.guardianmonsters.scene2d.SubImageImageButton;
import de.limbusdev.guardianmonsters.services.Services;

/**
 * GuardoSphereButton
 *
 * @author Georg Eckert 2017
 */

public class GuardoSphereButton extends SubImageImageButton {

    public GuardoSphereButton(Skin skin, AGuardian guardian) {

        super(skin, (guardian != null ? "button-gs" : "button-gs-empty"), construct(guardian, skin));
    }

    private static Image construct(AGuardian guardian, Skin skin) {

        TextureRegionDrawable drawable;

        if(guardian == null) {
            drawable = new TextureRegionDrawable(skin.getRegion("transparent"));
        } else {
            drawable = new TextureRegionDrawable(
                    Services.getMedia().getMonsterMiniSprite(guardian.getSpeciesDescription().getID(), guardian.getAbilityGraph().getCurrentForm()));
        }
        Image miniSprite = new Image(drawable);
        return miniSprite;
    }
}

File: GuardianMonsters/guardosphere/src/de/limbusdev/guardianmonsters/guardosphere/GuardoSphereButton.kt
/**
 * Copyright (C) 2019 Georg Eckert - All Rights Reserved
 */

package de.limbusdev.guardianmonsters.guardosphere

import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.ImageButton
import com.badlogic.gdx.scenes.scene2d.ui.Skin
import com.badlogic.gdx.scenes.scene2d.utils.Drawable
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable
import com.badlogic.gdx.utils.Align
import de.limbusdev.guardianmonsters.guardians.monsters.AGuardian
import de.limbusdev.guardianmonsters.scene2d.SubImageImageButton
import de.limbusdev.guardianmonsters.services.Services
import ktx.style.get

/**
 * GuardoSphereButton
 *
 * @author Georg Eckert 2017
 */

class GuardoSphereButton
(
        skin: Skin,
        guardian: AGuardian?
)
    : SubImageImageButton(skin, "button-gs-selection", construct(skin, guardian))
{
    companion object
    {
        private fun construct(skin: Skin, guardian: AGuardian?): Image
        {
            val drawable = when(guardian == null)
            {
                true  -> TextureRegionDrawable(skin.get<TextureRegion>("transparent"))
                false -> TextureRegionDrawable(Services.Media().getMonsterMiniSprite(guardian.speciesID, guardian.currentForm))
            }

            return Image(drawable)
        }
    }
}
